keyboards/wolf/xox70/hotswap/keymaps/via/keymap.c

```diff
@@ -0,0 +1,217 @@
+    xox.caps.s       = 128;
+    xox.caps.v       = RGB_MATRIX_MAXIMUM_BRIGHTNESS / 2;
+    xox.caps.enabled = true;
+    xox.leds         = LED_STATE_ALL;
+
+    // Write default value to EEPROM now
+    eeconfig_update_kb_datablock(&xox);
+}
+
+enum via_xox {
+    id_caps_indicator_enabled    = 1,
+    id_caps_indicator_brightness = 2,
+    id_caps_indicator_color      = 3,
+    id_leds_state                = 4
+};
+
+//On Keyboard startup
+void keyboard_post_init_user(void) {
+    //Read our custom menu variables from memory
+    eeconfig_read_kb_datablock(&xox);
+}
+
+void set_caps_lock_led(void) {
+    if (xox.caps.enabled) {
+        // The rgb_matrix_set_color function needs an RGB code to work, so first the indicator color is cast to an HSV value and then translated to RGB
+        HSV hsv_caps_indicator_color = {xox.caps.h, xox.caps.s, xox.caps.v};
+        RGB rgb_caps_indicator_color = hsv_to_rgb(hsv_caps_indicator_color);
+        if (host_keyboard_led_state().caps_lock)
+            rgb_matrix_set_color(CAPS_INDICATOR_INDEX, rgb_caps_indicator_color.r, rgb_caps_indicator_color.g, rgb_caps_indicator_color.b);
+    }
+}
+
+bool rgb_matrix_indicators_advanced_user(uint8_t led_min, uint8_t led_max) {
+
+    switch(xox.leds){
+            case LED_STATE_ALL:
+                set_caps_lock_led();
+                return true;
+            case LED_STATE_BACKLIGHT:
+                for (uint8_t i = led_min; i < led_max; i++) {
+                    if (!HAS_FLAGS(g_led_config.flags[i], LED_FLAG_KEYLIGHT)) {
+                        rgb_matrix_set_color(i, 0, 0, 0);
+                    }
+                }
+                set_caps_lock_led();
+                return true;
+            case LED_STATE_UNDERGLOW: 
+                for (uint8_t i = led_min; i < led_max; i++) {
+                    if (!HAS_FLAGS(g_led_config.flags[i], LED_FLAG_UNDERGLOW)) {
+                        rgb_matrix_set_color(i, 0, 0, 0);
+                    }
+                }
+                set_caps_lock_led();
+                return true;
+            case LED_STATE_INDICATORS:
+                for (uint8_t i = led_min; i < led_max; i++) {
+                    rgb_matrix_set_color(i, 0, 0, 0);
+                }
+                set_caps_lock_led();
+                return true;
+        }
+
+    return true;
+}
+
+void xox_config_set_value(uint8_t *data) {
+    // data = [ value_id, value_data ]
+    uint8_t *value_id   = &(data[0]);
+    uint8_t *value_data = &(data[1]);
+
+    switch (*value_id) {
+        case id_caps_indicator_enabled: {
+            xox.caps.enabled = value_data[0];
+            break;
+        }
+        case id_caps_indicator_brightness: {
+            xox.caps.v = value_data[0];
+            break;
+        }
+        case id_caps_indicator_color: {
+            xox.caps.h = value_data[0];
+            xox.caps.s = value_data[1];
+            break;
+        }
+        case id_leds_state: {
+            xox.leds = value_data[0];
```

Comment:
This could and probably should be translated to use the rgb_matrix_set_flags function.

Reply:
I've removed the configuration from the VIA RGB to be more compliant with current QMK's stance on too much code in the keymaps that's duplicate to many other boards.
I appreciate you looking at this